src/PIL/ImageFilter.py

```diff
@@ -297,3 +299,150 @@ class SMOOTH_MORE(BuiltinFilter):
         1,  5,  5,  5,  1,
         1,  1,  1,  1,  1
         )
+
+
+class Color3DLUT(MultibandFilter):
+    """Three-dimensional color lookup table.
+
+    Transforms 3-channel pixels using the values of the channels as coordinates
+    in the 3D lookup table and interpolating the nearest elements.
+
+    This method allows you to apply almost any color transformation
+    in constant time by using pre-calculated decimated tables.
+
+    :param size: Size of the table. One int or tuple of (int, int, int).
+                 Minimal size in any dimension is 2, maximum is 65.
+    :param table: Flat lookup table. A list of ``channels * size**3``
+                  float elements or a list of ``size**3`` channels-sized
+                  tuples with floats. Channels are changed first,
+                  then first dimension, then second, then third.
+                  Value 0.0 corresponds lowest value of output, 1.0 highest.
+    :param channels: Number of channels in the table. Could be 3 or 4.
+                     Default is 3.
+    :param target_mode: A mode for the result image. Should have not less
+                        than ``channels`` channels. Default is ``None``,
+                        which means that mode wouldn't be changed.
+    """
+    name = "Color 3D LUT"
+
+    def __init__(self, size, table, channels=3, target_mode=None):
+        self.size = size = self._check_size(size)
+        self.channels = channels
+        self.mode = target_mode
+
+        table = list(table)
+        # Convert to a flat list
+        if table and isinstance(table[0], (list, tuple)):
+            table, raw_table = [], table
+            for pixel in raw_table:
+                if len(pixel) != channels:
+                    raise ValueError("The elements of the table should have "
+                                     "a length of {}.".format(channels))
+                for color in pixel:
+                    table.append(color)
+
+        if len(table) != channels * size[0] * size[1] * size[2]:
+            raise ValueError(
+                "The table should have channels * size**3 float items "
+                "either size**3 items of channels-sized tuples with floats. "
+                "Table size: {}x{}x{}. Table length: {}".format(
+                    size[0], size[1], size[2], len(table)))
+        self.table = table
+
+    @staticmethod
+    def _check_size(size):
+        try:
+            _, _, _ = size
+        except ValueError:
+            raise ValueError("Size should be an integer either "
+                             "tuple of three integers.")
```

Comment:
I presume you meant 'Size should be either an integer or a tuple of three integers'.

Although I'm not convinced they need to be integers, since you convert the values to integers in the next line after this block?

Reply:
With duck typing, "should be an integer" means that it should be something that can be used as an integer. So I'm converting values to check that they are really integers or can be converted to int.